File: src/main/java/com/github/hhhzzzsss/songplayer/commands/TestSongCommand.java
package com.github.hhhzzzsss.songplayer.commands;

import com.github.hhhzzzsss.songplayer.playing.NotePlayer;
import com.github.hhhzzzsss.songplayer.song.Note;
import com.github.hhhzzzsss.songplayer.song.Song;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;

class TestSongCommand extends Command {
    @Override
    public String getName() {
        return "testSong";
    }

    @Override
    public String getDescription() {
        return "Creates a song for testing";
    }

    @Override
    public void buildNode(LiteralArgumentBuilder<FabricClientCommandSource> node) {
        node.executes(context -> {
            Song song = new Song("test_song");
            for (int i=0; i<400; i++) {
                song.add(new Note(i, i*50));
            }
            song.length = 400*50;
            NotePlayer.instance.setSong(song);

            return 1;
        });
    }
}

File: src/main/java/com/github/hhhzzzsss/songplayer/commands/TestSongCommand.kt
package com.github.hhhzzzsss.songplayer.commands

import com.github.hhhzzzsss.songplayer.playing.SongHandler
import com.github.hhhzzzsss.songplayer.song.Note
import com.github.hhhzzzsss.songplayer.song.Song
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource

internal class TestSongCommand : Command {
    override val name = "testSong"
    override val description = "Creates a song for testing"

    override fun buildNode(node: LiteralArgumentBuilder<FabricClientCommandSource>) {
        node.executes {
            val song = Song("Test song")
            for (i in 0..399) {
                song.add(Note(i, (i * 50).toLong()))
            }
            song.length = (400 * 50).toLong()
            SongHandler.instance.songQueue.forcePlay(song)
            1
        }
    }
}